src/v/kafka/server/connection_context.cc

```diff
@@ -372,14 +378,20 @@ ss::future<> connection_context::consume_stages(
     ssx::background
       = ssx::spawn_with_gate_then(
           _rs.conn_gate(),
-          [this, f = std::move(f), seq, correlation]() mutable {
-              return f.then([this, seq, correlation](response_ptr r) mutable {
+          [this,
+           f = std::move(f),
+           record_time_quota,
+           seq,
+           correlation]() mutable {
+              return f.then([this, record_time_quota, seq, correlation](
+                              response_ptr r) mutable {
+                  record_time_quota();
```

Comment:
Does the fact that this is happening in the background mean that the recorded time for the request isn't necessarily representative of CPU time taken by this request only?

Reply:
Right, and since we're only capturing the start time, things like waiting on IO, waiting in a seastar queue, etc. are included in the measurement. My point is that we're not only measuring CPU cycles taken by this specific client request, but a few other things. High values for all of these are still indicative that the client should probably be throttled, but just wanted to confirm that e.g. a given window can far exceed 100% of the quota.